search/query-features-with-arcade-expression/src/main/java/com/esri/samples/query_features_with_arcade_expression/QueryFeaturesWithArcadeExpressionSample.java

```diff
@@ -0,0 +1,198 @@
+  public void start(Stage stage) {
+    try {
+      // create stack pane and application scene
+      var stackPane = new StackPane();
+      var scene = new Scene(stackPane);
+
+      // set title, size, and add scene to stage
+      stage.setTitle("Query Features With Arcade Expression Sample");
+      stage.setWidth(800);
+      stage.setHeight(700);
+      stage.setScene(scene);
+      stage.show();
+
+      // authentication with an API key or named user is required to access basemaps and other location services
+      String yourAPIKey = System.getProperty("apiKey");
+      ArcGISRuntimeEnvironment.setApiKey(yourAPIKey);
+
+      // create a portal item and use it to create a map
+      var portal = new Portal("https://www.arcgis.com/");
+      var portalItem = new PortalItem(portal, "539d93de54c7422f88f69bfac2aebf7d");
+      map = new ArcGISMap(portalItem);
+
+      // create a map view and set the map to it
+      mapView = new MapView();
+      mapView.setMap(map);
+
+      // set callout properties that never change
+      mapView.getCallout().setLeaderPosition(Callout.LeaderPosition.BOTTOM);
+      mapView.getCallout().setTitle("RPD Beats");
+
+      // wait for the map to finish loading data from portal before accessing its layers
+      map.addDoneLoadingListener(() -> {
+        if (map.getLoadStatus() == LoadStatus.LOADED) {
+          // hide all layers other than the RPD Beats Layer to reduce clutter
+          map.getOperationalLayers().forEach(layer -> layer.setVisible(layer.getName().equals(rpdBeatsLayerName)));
+
+          // get the RPD beats layer.
+          policeBeatsLayer = map.getOperationalLayers().stream().filter(
+            layer -> layer.getName().equals(rpdBeatsLayerName)).toList().get(0);
+
+          // if map clicked, evaluate an arcade expression at that point
+          mapView.addEventHandler(MouseEvent.MOUSE_CLICKED, event -> {
+            if (event.getButton() == MouseButton.PRIMARY && event.isStillSincePress()) {
+              try {
+                evaluateArcadeExpression(new Point2D(event.getX(), event.getY()));
+              } catch (Exception e) {
+                throw new RuntimeException(e);
+              }
+            }});
+        } else if (map.getLoadStatus() == LoadStatus.FAILED_TO_LOAD) {
+          new Alert(Alert.AlertType.ERROR, "Map failed to load").show();
+        }
+      });
+
+      // add the map view to the stack pane
+      stackPane.getChildren().addAll(mapView);
+    } catch (Exception e) {
+      // on any error, display the stack trace.
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Evaluates an arcade expression at the point on the map that was clicked.
+   *
+   * @param point The point on the map which was clicked, in screen (rather than map) coordinates.
+   * @throws RuntimeException If unable to get results from futures due to interruption or concurrent execution.
+   */
+  private void evaluateArcadeExpression(Point2D point) throws RuntimeException {
```

Comment:
the contents of this whole method can just go in place where it's called above. We don't have any repeated code here and it's more of a story flow, so it's ok to have it in place.